### ISSUE TITLE
Add protocol header for upstream service

### ISSUE THREAD

**Issue by chnliyong** [apache/incubator-apisix#973]
The upstream service such as web rendering may sometimes need to know the incoming request's scheme(`http` or `https`) from client. 
Conventionally use the http header `X-Forwarded-Proto`[X-Forwarded-Proto](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto)

**Comment by moonming** [comment]
I think you can use the proxy-rewrite plugin.

**Comment by chnliyong** [comment]
> I think you can use the proxy-rewrite plugin.

I thought this is a very commonly requirement if `apisix` is used as a `reverse proxy` to offer ssl offloading, and configure the `proxy-rewrite`  for all routes is tedious.
We may add a option such as `enable_protocol_forwarded` and `protocol_forwarded_header` in `config.yaml` if we don't  want this as default.  Actually, I would like this be default.

```nginx
      proxy_set_header ....
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header ....
```

**Comment by moonming** [comment]
I got your point.
I think we can add this to `/bin/apisix` directly.

**Comment by moonming** [comment]
you can send email to mailing list to discuss with more developer.

**Comment by moonming** [comment]
@chnliyong any update?

**Comment by agile6v** [comment]
@chnliyong 
Hi, will you continue to follow up on this issue?

**Comment by agile6v** [comment]
This issue has been fixed by #1113 .
